tests/integration/targets/aci_hsrp_interface_policy/tasks/main.yml

```
@@ -0,0 +1,154 @@
+      <<: *tenant_absent
+      state: present
+
+  - name: Add a HSRP interface policy in check mode
+    cisco.aci.aci_hsrp_interface_policy: &hsrp_present
+      <<: *tenant_present
+      hsrp: hsrp1
+      controls: ["bia"]
+      reload_delay: 2
+      delay: 2
+      state: present
+    check_mode: true
+    register: cm_add_hsrp
+
+  - name: Add a HSRP interface policy in normal mode
+    cisco.aci.aci_hsrp_interface_policy: *hsrp_present
+    register: nm_add_hsrp
+
+  - name: Verify add hsrp
+    assert:
+      that:
+      - cm_add_hsrp is changed
+      - nm_add_hsrp is changed
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.ctrl == "bia"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.delay == "2"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.reloadDelay == "2"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.name == "hsrp1"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.dn == "uni/tn-ansible_test/hsrpIfPol-hsrp1"
+
+  - name: Add the HSRP interface policy again to check idempotency
+    cisco.aci.aci_hsrp_interface_policy: *hsrp_present
+    register: nm_add_hsrp_again
+
+  - name: Verify add_hsrp again
+    assert:
+      that:
+      - nm_add_hsrp_again is not changed
+
+  - name: Update hsrp interface policy
+    cisco.aci.aci_hsrp_interface_policy: &hsrp_update
+      <<: *hsrp_present
+      delay: 3
+    register: nm_hsrp_update
+
+  - name: Verify update hsrp
+    assert:
+      that:
+      - nm_hsrp_update is changed
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.delay == "3"
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.reloadDelay == "2"
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.name == "hsrp1"
+      - nm_hsrp_update.current.0.hsrpIfPol.attributes.dn == "uni/tn-ansible_test/hsrpIfPol-hsrp1"
+
+  - name: Update hsrp interface policy to check for errors in delay
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *hsrp_update
+      delay: 10001
+    register: nm_hsrp_error1
+    ignore_errors: true
+
+  - name: Update hsrp interface policy to check for errors in reload delay
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *hsrp_update
+      reload_delay: 10001
+    register: nm_hsrp_error2
+    ignore_errors: true
+
+  - name: Verify errors
+    assert:
+      that:
+      - nm_hsrp_error1.msg == "Parameter 'delay' is only valid in range between 1 and 10000."
+      - nm_hsrp_error2.msg == "Parameter 'reload_delay' is only valid in range between 1 and 10000."
+
+  - name: Add another HSRP interface policy in check mode
+    cisco.aci.aci_hsrp_interface_policy: &hsrp2
+      <<: *tenant_present
+      hsrp: hsrp2
+      controls: ["bfd"]
+      state: present
+    register: add_hsrp2
+
+  - name: Query hsrp interface policy 1
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *hsrp_update
+      state: query
+    register: query_hsrp
+
+  - name: Query all hsrp interface policies
+    cisco.aci.aci_hsrp_interface_policy:
+      <<: *tenant_present
```

should query all not be without input variables, thus *aci_info shoudl be used to trigger the class query?
```suggestion
      <<: *aci_info
```

Comment applies for other modules also.

---

Ok so class querying is then not supported for this module? Because in my understanding that would be the part we want to test with query all? Else it would be "Query all hsrp interface policies for tenant".

Also you can test >= to the number of hsrp_interface_policies in the query all to test. Then if you also want to add for tenant filtering it should in my opinion be defined as a separate test. 

If we are doing the tenant filtering test, for consistency you likely also want to define tests for every level of nesting / filtering, that the get request is created with the correct filter.  

Both scenarios are not something we usually test as far as I can see, where we typically do the most nested level of query ( query_one ) and a generic class query ( query_all ). So personally I would prefer keeping it consistent with other modules and do the class query and the filtered query.